firewood/src/db.rs

```diff
@@ -631,13 +612,28 @@ impl Db {
             &cfg.rev,
         )?;
 
+        let wal_config = WalConfig::builder()
+            .file_nbit(params.wal_file_nbit)
+            .block_nbit(params.wal_block_nbit)
+            .max_revisions(cfg.wal.max_revisions)
+            .build();
+
+        #[allow(clippy::unwrap_used)]
+        let disk_buffer =
+            DiskBuffer::new(inbound, &cfg.buffer, &wal_config).expect("DiskBuffer::new");
+
         Ok(Self {
             inner: Arc::new(RwLock::new(DbInner {
-                disk_thread,
+                disk_thread: Some(
+                    std::thread::Builder::new()
+                        .name("DiskBuffer".to_string())
+                        .spawn(move || disk_buffer.run())
+                        .expect("thread spawn should succeed"),
+                ),
                 disk_requester,
                 cached_space: data_cache,
-                reset_store_headers: reset_headers,
-                root_hash_staging,
+                reset_store_headers: reset,
```

Comment:
nit: if we just name these the same, we can skip the assignment. Just call this `reset_store_headers`. Or, since this is likely the only caller, change the name of the argument.